pyg_lib/csrc/ops/cpu/matmul_kernel.cpp

```diff
@@ -31,16 +278,18 @@ at::Tensor segment_matmul_kernel(const at::Tensor& input,
                                  const at::Tensor& other) {
   const auto size = pyg::utils::size_from_ptr(ptr).cpu();
   const auto sizes = at::IntArrayRef(size.data_ptr<int64_t>(), size.numel());
-  const auto out = input.new_empty({input.size(0), other.size(-1)});
+  const auto input_contig = input.contiguous();
+  const auto other_contig = other.contiguous();
+  const auto out = input_contig.new_empty({input.size(0), other.size(-1)});
 
   auto outs = out.split_with_sizes(/*split_size=*/sizes, /*dim=*/0);
   for (auto& out_part : outs) {
     out_part.resize_(0);
   }
 
   grouped_matmul_out_kernel(
-      input.contiguous().split_with_sizes(/*split_size=*/sizes, /*dim=*/0),
-      other.contiguous().split(/*split_size=*/1, /*dim=*/0), outs);
+      input_contig.split_with_sizes(/*split_size=*/sizes, /*dim=*/0),
```

Comment:
`split_with_sizes` will create new tensor views, again it is `at::Tensor` constructor issue.
Notice that this is a very heavy struct, we should avoid construct/deconstruct heavy struct as much as possible.

Reply:
Good point 😉 I think I can get rid of `split_with_sizes` in the MKL execution path by changing leading dimensions.